tests/test_utils.py

```diff
@@ -327,7 +327,7 @@ def test_get_compile_command_sort_args(tmpdir_cwd):
         pass
 
     args = [
-        "--no-index",
```

Comment:
I think that during the deprecation period, the old flag should still be tested and behave like the new one.

Reply:
It's tested in `test_index_option`. The `test_get_compile_command_sort_args ` is not about `--index`'s behaviour actually, it tests sorting args in `get_compile_command()`.